### PR TITLE
Wrap methods

### PR DESCRIPTION
**Pull request by lukedirtwalker**
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/dimitar-asenov/Envision/pull/195%23discussion_r43253330%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/195%23issuecomment-151843474%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/195%23discussion_r43253790%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/195%23issuecomment-151844194%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/195%23issuecomment-151845397%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/195%23issuecomment-151843474%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Do%20we%20have%20any%20script%20which%20tries%20to%20use%20this%20new%20stuff%3F%20Do%20we%20have%20at%20least%20some%20evidence%20that%20it%20does%20the%20right%20thing%3F%22%2C%20%22created_at%22%3A%20%222015-10-28T13%3A22%3A19Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%2C%20%7B%22body%22%3A%20%22Hm%20I%20will%20use%20this%20stuff%20for%20refactoring%20scripts.%20I%20have%20just%20written%20a%20dummy%20script%20which%20renames%20all%20methods%20of%20a%20class%20to%20foo%20by%20using%3A%5Cr%5Cn%60%60%60%5Cr%5Cn%20%20%20%20method.beginModification%28%5C%22rename%5C%22%29%5Cr%5Cn%20%20%20%20method.name%20%3D%20%5C%22Foo%5C%22%5Cr%5Cn%20%20%20%20method.endModification%28%29%5Cr%5Cn%60%60%60%5Cr%5Cnand%20that%20at%20least%20works%20nicely.%5Cr%5Cn%5Cr%5CnOn%20the%20other%20hand%20testing%20everything%20could%20be%20quite%20difficult..%22%2C%20%22created_at%22%3A%20%222015-10-28T13%3A25%3A39Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3530769%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/lukedirtwalker%22%7D%7D%2C%20%7B%22body%22%3A%20%22Well%2C%20I%20didn%27t%20really%20mean%20to%20test%20everything%2C%20just%20whether%20the%20general%20generation%20and%20python%20usage%20works%2C%20which%20you%27ve%20done%2C%20so%20I%27ll%20merge%20it.%22%2C%20%22created_at%22%3A%20%222015-10-28T13%3A31%3A10Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20b00393dda0ffb2eaaf84534debcf14b760c3a14d%20PythonWrapperGenerator/src/EnvisionAstConsumer.cpp%2010%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/195%23discussion_r43253330%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Here%20you%20say%20you%20ignore%20overloaded%20methods%2C%20but%20below%20you%20ignore%20overloaded%20operators.%20Which%20one%20is%20it%3F%22%2C%20%22created_at%22%3A%20%222015-10-28T13%3A19%3A43Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%2C%20%7B%22body%22%3A%20%22should%20be%20operators%22%2C%20%22created_at%22%3A%20%222015-10-28T13%3A23%3A33Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3530769%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/lukedirtwalker%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20PythonWrapperGenerator/src/EnvisionAstConsumer.cpp%3AL125-154%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull b00393dda0ffb2eaaf84534debcf14b760c3a14d PythonWrapperGenerator/src/EnvisionAstConsumer.cpp 10'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/195#discussion_r43253330'>File: PythonWrapperGenerator/src/EnvisionAstConsumer.cpp:L125-154</a></b>
- <a href='https://github.com/dimitar-asenov'></a> Here you say you ignore overloaded methods, but below you ignore overloaded operators. Which one is it?
- <a href='https://github.com/lukedirtwalker'></a> should be operators
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/195#issuecomment-151843474'>General Comment</a></b>
- <a href='https://github.com/dimitar-asenov'></a> Do we have any script which tries to use this new stuff? Do we have at least some evidence that it does the right thing?
- <a href='https://github.com/lukedirtwalker'></a> Hm I will use this stuff for refactoring scripts. I have just written a dummy script which renames all methods of a class to foo by using:

```
method.beginModification("rename")
method.name = "Foo"
method.endModification()
```

and that at least works nicely.
On the other hand testing everything could be quite difficult..
- <a href='https://github.com/dimitar-asenov'></a> Well, I didn't really mean to test everything, just whether the general generation and python usage works, which you've done, so I'll merge it.

<a href='https://www.codereviewhub.com/dimitar-asenov/Envision/pull/195?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/dimitar-asenov/Envision/pull/195?approve=1'></a>&nbsp;<a href='https://github.com/dimitar-asenov/Envision/pull/195'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dimitar-asenov** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/dimitar-asenov/Envision/pull/195'>Refresh</a>.<a href='#crh-update-none'></a>

File: PythonWrapperGenerator/src/EnvisionAstConsumer.cpp [hunk 1]
```diff
@@ -125,15 +125,29 @@ void EnvisionAstConsumer::HandleClassDecl(clang::CXXRecordDecl* classDecl)
 						possibleAttributeSetters << methodName;
 				}
 			}
+			QMultiHash<QString, clang::CXXMethodDecl*> overloads;
 			// check if we have some more attributes which don't have an attribute macro:
 			for (auto method : classDecl->methods())
 			{
-				// Ignore private methods:
-				if (method->getAccess() == clang::AccessSpecifier::AS_private) continue;
+				// Ignore de-/constructors & deleted methods & overloaded methods:
```
Comment (dimitar-asenov) [inline PythonWrapperGenerator/src/EnvisionAstConsumer.cpp]: Here you say you ignore overloaded methods, but below you ignore overloaded operators. Which one is it?

**Comment by dimitar-asenov** [comment]
Do we have any script which tries to use this new stuff? Do we have at least some evidence that it does the right thing?

File: PythonWrapperGenerator/src/EnvisionAstConsumer.cpp (hunk 1 above)
Comment (lukedirtwalker) [inline PythonWrapperGenerator/src/EnvisionAstConsumer.cpp]: should be operators

**Comment by lukedirtwalker** [comment]
Hm I will use this stuff for refactoring scripts. I have just written a dummy script which renames all methods of a class to foo by using:

```
    method.beginModification("rename")
    method.name = "Foo"
    method.endModification()
```

and that at least works nicely.

On the other hand testing everything could be quite difficult..

**Comment by dimitar-asenov** [comment]
Well, I didn't really mean to test everything, just whether the general generation and python usage works, which you've done, so I'll merge it.

**Pull request merged**